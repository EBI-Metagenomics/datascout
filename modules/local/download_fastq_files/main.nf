#!/usr/bin/env nextflow
process DOWNLOAD_FASTQ_FILES {

    container 'community.wave.seqera.io/library/pigz_python_pip_boto3_pandas:b10b66ffdecc53f1'
    debug true
    publishDir "${params.output}", mode: "copy"

    errorStrategy 'retry'
    maxRetries 2

    input:
      tuple val(meta), file(ena_metadata)
      val(start_line)

    output:
      tuple val(meta), path("${meta.id}_rna_fastq_dir"), path("${meta.id}_rna_fastq_dir/*"), emit: fastq_files

    script:
    """
    download_rnaseq_fastqs.py --startline ${start_line} --transcriptomes ${ena_metadata} --output-dir "${meta.id}_rna_fastq_dir"
    """
}

// Download subset of fastq files for 5 accessions
