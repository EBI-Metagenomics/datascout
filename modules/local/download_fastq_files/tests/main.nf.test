nextflow_process {

    name "Test Process DOWNLOAD_FASTQ_FILES"
    script "modules/local/download_fastq_files/main.nf"
    process "DOWNLOAD_FASTQ_FILES"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = tuple([id:"GCA_003719455.1", ena_tax:"default"], "$projectDir/tests/reference_databases/GCA_003719455.1_ENA_filtered_rna.csv")
                input[1] = 1
                input[2] = 1
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.fastq_files).match()
        }

    }

}
