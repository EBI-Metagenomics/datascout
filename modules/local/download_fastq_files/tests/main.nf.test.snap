{
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "id": "GCA_003719455.1",
                        "ena_tax": "default"
                    },
                    [
                        "SRR25010087_1.fastq:md5,717e8d1c6558ef61a60b08296f4c8b3a",
                        "SRR25010087_2.fastq:md5,c53cfbdc10ea4bde941845452ab5c2b9"
                    ]
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-09-10T16:45:43.350895"
    }
}